src/Action/Task/GenerateOrders.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace AppBundle\Action\Task;
+
+use AppBundle\Entity\Sylius\Order;
+use AppBundle\Entity\Task;
+use AppBundle\Pricing\PricingManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Recurr\Transformer\ArrayTransformer;
+use Recurr\Transformer\Constraint\BetweenConstraint;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+class GenerateOrders
+{
+    public function __construct(
+        private readonly EntityManagerInterface $entityManager,
+        private readonly PricingManager $pricingManager)
+    {
+    }
+
+    public function __invoke($data, Request $request)
+    {
+        //get query parameters
+        $queryParams = $request->query->all();
+
+        $date = $queryParams['date'];
+
+        if (empty($date)) {
+            throw new BadRequestHttpException('Date is required');
+        }
+
+        $allSubscriptions = $this->entityManager->getRepository(Task\RecurrenceRule::class)->findBy(['deletedAt' => null]);
```

Comment:
same we should move RecurrenceRule outside of Task "sub entity" I think

Reply:
Yes! I'll do it in the next version